From 74bb3071c39504d07583648e8f80ca99a876d668 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 22 Sep 2017 22:37:24 +0800
Subject: [PATCH 2584/5242] MLK-16536-10 video: fbdev: dcss: unmask
 'IRQ_TC_LINE1' by default

commit  b13c5586c46a95e7f4e3122f6ab0b3ade53483a4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Unmask the 'IRQ_TC_LINE1' when initialize it by default,
since the vsync count can be used as a reference count or
timestamp.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 5b54c14..584bfa9 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2829,7 +2829,6 @@ static int dcss_wait_for_vsync(unsigned long crtc,
 	struct platform_device *pdev = info->pdev;
 
 	spin_lock_irqsave(&info->vinfo.vwait.lock, irqflags);
-	dtg_irq_unmask(IRQ_TC_LINE1, info);
 	vcount = info->vinfo.vcount;
 	spin_unlock_irqrestore(&info->vinfo.vwait.lock, irqflags);
 
@@ -2931,8 +2930,9 @@ static irqreturn_t dcss_irq_handler(int irq, void *dev_id)
 		dtg_irq_clear(IRQ_TC_LINE1, info);
 
 		spin_lock_irqsave(&info->vinfo.vwait.lock, irqflags);
+
 		info->vinfo.vcount++;
-		dtg_irq_mask(IRQ_TC_LINE1, info);
+
 		spin_unlock_irqrestore(&info->vinfo.vwait.lock, irqflags);
 
 		wake_up_all(&info->vinfo.vwait);
@@ -2966,6 +2966,7 @@ static int dcss_interrupts_init(struct dcss_info *info)
 		case 8:		/* dtg_programmable_1: for vsync */
 			/* TODO: (0, 0) or (last, last)? */
 			writel(0x0, info->base + chans->dtg_addr + TC_LINE1_INT);
+			dtg_irq_unmask(IRQ_TC_LINE1, info);
 			break;
 		default:	/* TODO: add support later */
 			continue;
-- 
1.7.9.5

